text/0000-embedded-refract.md

```diff
@@ -0,0 +1,361 @@
+          "element": "string",
+          "content": "scores"
+        },
+        "value": {
+          "element": "array",
+          "content": [
+            {
+              "element": "number",
+              "content": 150
+            },
+            {
+              "element": "number",
+              "content": 202
+            },
+            {
+              "element": "number",
+              "content": 145
+            }
+          ]
+        }
+      }
+    }
+  ]
+}
+```
+
+## Refracting Value Elements
+
+If values do not have any meta values or attributes, there is no need to represent them as refracted. If I were to add some annotation to one of the values, it would show up as Refract and would be defined as Refracted by use of the `refract` property keyword.
+
+```json
+{
+  "first_name": "John",
+  "last_name": "Doe",
+  "age": {
+    "refract": {
+      "element": "number",
+      "meta": {
+        "id": "bowner-103-age"
+      },
+      "content": 28
+    }
+  },
+  "scores": [150, 202, 145]
+}
+```
+
+Above, I've added an ID to the age of the bowler. Notice how I just refracted that value in that specific place. This is also the same as:
+
+```json
+{
+  "refract": {
+    "element": "object",
+    "meta": {
+      "id": "bowler-103"
+    },
+    "content": {
+      "age": {
+        "element": "number",
+        "meta": {
+          "id": "bowner-103-age"
+        },
+        "content": 28
+      }
+    }
+  },
+  "first_name": "John",
+  "last_name": "Doe",
+  "scores": [150, 202, 145]
+}
+```
+
+This will also support full member elements, so even this below is equivalent (note that it's a slight modification from the full Refract):
+
+```json
+{
+  "refract": {
+    "element": "object",
+    "meta": {
+      "id": "bowler-103"
+    },
+    "content": [
+      {
+        "element": "member",
+        "content": {
+          "key": "age",
+          "value": {
+            "refract": {
+              "element": "number",
+              "meta": {
+                "id": "bowner-103-age"
+              },
+              "content": 28
+            }
+          }
+        }
+      }
+    ]
+  },
+  "first_name": "John",
+  "last_name": "Doe",
+  "scores": [150, 202, 145]
+}
+```
+
+In full Refract, you would have to also provide the element for the key "age" in the previous example, but since we can rely on the `refract` keyword, we can leave it unrefracted. We could Refract it more later if necessary.
+
+## Refracting Array and Array Items
+
+Array items can be embedded without refracting the entire array.
+
+```json
+{
+  "first_name": "John",
+  "last_name": "Doe",
+  "age": 28,
+  "scores": [
+    {
+      "refract": {
+        "element": "number",
+        "meta": {
+          "id": "bowler-103-game-1"
+        },
+        "content": 150
+      }
+    },
+    202,
+    145]
+}
+```
+
+Full arrays can also be refracted.
+
+```json
+{
+  "first_name": "John",
+  "last_name": "Doe",
+  "age": 28,
+  "scores": {
```

Comment:
I actually had a branch that handled this, and it wasn't bad. The only thing you do is check every object for `_refract`. If it's there, you convert to element differently. So:

- If object, check for `_refract`
- If `_refract` is there, use special instructions for converting the Refract Object to element instance
- If not there, treat as normal JSON and convert to literal element instance